### PR TITLE
feat(APP-2928): Add reset filter functionality to DataListFilter core component

### PR DESCRIPTION
**Pull request by shan8851**
## Description

- Add functionality to reset all filters in `DataListFilter` core component.
- Add additional tests to ensure button is rendered correctly (when status === 'filtered') and to ensure `onResetFiltersClick` callback is called correctly.

Task: [APP-2929](https://aragonassociation.atlassian.net/browse/APP-2928)

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

-   [x] I have selected the correct base branch.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [ ] Any dependent changes have been merged and published in downstream modules.
-   [x] I ran all tests with success and extended them if necessary.
-   [x] I have updated the `CHANGELOG.md` file in the root folder of the package after the [UPCOMING] title and before
        the latest version.
-   [ ] I have tested my code on the test network.

### CODE REVIEW

File: src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx [hunk 1]
```diff
@@ -20,28 +23,35 @@ export const DataListFilterStatus: React.FC<IDataListFilterStatusProps> = () =>
...
+                {isLoading && copy.dataListFilterStatus.filteringEntity(entityLabel)}
+                {isFiltered && (
+                    <>
+                        <span>{copy.dataListFilterStatus.found} </span>
+                        <span className="text-primary-400">{itemsCount} </span>
+                        <span>{entityLabel}</span>
+                    </>
+                )}
+            </p>
             {isFiltered && (
```
Comment (cgero-eth) [inline src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx:50]: We should also check that the `onResetFiltersClick` property is defined. I would also add a test to make sure that the button is not rendered when `onResetFiltersClick` is not defined

File: src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx [hunk 2]
```diff
@@ -20,28 +23,35 @@ export const DataListFilterStatus: React.FC<IDataListFilterStatusProps> = () =>
...
-        >
-            {displayItemsCount && (
-                <>
-                    <span className="text-neutral-800">{itemsCount} </span>
-                    <span>{entityLabel}</span>
-                </>
-            )}
-            {isInitialLoading && copy.dataListFilterStatus.loadingEntity(entityLabel)}
-            {isLoading && copy.dataListFilterStatus.filteringEntity(entityLabel)}
+        <div className="flex h-10 items-center justify-between">
```
Comment (cgero-eth) [inline src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx:26]: We don't probably need to set a fixed height here
```suggestion
        <div className="flex items-center justify-between">
```

File: src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx [hunk 3]
```diff
@@ -1,10 +1,13 @@
 import classNames from 'classnames';
+import { Button } from '../../button';
 import { useOdsCoreContext } from '../../odsCoreProvider';
 import { useDataListContext } from '../dataListContext';
 
-export interface IDataListFilterStatusProps {}
+export interface IDataListFilterStatusProps {
+    onResetFiltersClick?: () => void;
```
Comment (cgero-eth) [inline src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx:7]: Could you add a short description for this property?

**Review by cgero-eth — COMMENTED** [review]
Thanks for the changes @shan8851!

File: src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx (hunk 2 above)
Comment (shan8851) [reply src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx:26]: @cgero-eth problem with not setting height is the content on the left is pushed down as soon as state transitions to `isFiltered` and the UI jumps if that makes sense.

File: src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx (hunk 1 above)
Comment (shan8851) [reply src/core/components/dataList/dataListFilter/dataListFilterStatus.tsx]: Good catch - thanks

**Review by cgero-eth — APPROVED** [review]
LGTM, thanks @shan8851!

**Pull request merged**